### ISSUE TITLE
[Dialog]: Adding tabindex breaks tabbing

### ISSUE THREAD

**Issue by bengeorge** [radix-ui/primitives#1673]
### Current Behavior

If you add a `tabindex` to an element in a dialog, tabbing is stuck on the last element that has a `tabindex` instead of going back to the next element in the tab order (first element with 0 / no `tabindex`).

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

> tabindex="0" means that the element should be focusable in sequential keyboard navigation, after any positive tabindex values and its order is defined by the document's source order.

....

> If not specified, it takes the default value 0.

### Expected behavior

Should be able to tab through all elements when a `tabindex` is in use.

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/cool-bohr-li53bq?file=/App.js:4539-4552)

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |         |    1.0.0     |
| React            | n/a     |    17.0.2     |
| Browser          |         |    Chrome     |
| Assistive tech   |         |         |
| Node             | n/a     |    16.16     |
| npm/yarn         |         |         |
| Operating System |         |    Windows     |

**Comment by bengeorge** [comment]
In general we try not to use `tabindex`, in a "large" application we use it in ~5 places. So isn't a massive blocker but from a pure HTML point of view I think it should still work.

**Comment by benoitgrelard** [comment]
Hi @bengeorge,

Yeah, we specifically do not handle positive `tabindex` as it would be very complex and largely hinders accessibility to have tab order different from visual order.

See here:
https://github.com/radix-ui/primitives/blob/83a8c13bf66f3d9f17d77caeb187a69eb146930b/packages/react/focus-scope/src/FocusScope.tsx#L230-L231

I will close the issue as "won't fix" as it isn't worth the added complexity and it largely works still (trapping) only looping doesn't work.

**Comment by benoitgrelard** [comment]
@ericmatthys that's a good suggestion, I've added an issue for it 👍
